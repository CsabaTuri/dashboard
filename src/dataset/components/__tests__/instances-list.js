import React from "react"
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"

import InstancesList from "../instances-list"

describe("InstancesList Component", () => {
  let comp, onSelectRef

  beforeEach(() => {
    onSelectRef = jest.fn()
    comp = shallow(<InstancesList query={jest.fn()} onSelectRef={onSelectRef} />)
  })

  it("displays instances list", () => {
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("displays the selected instance", () => {
    onSelectRef.mockReturnValue(Promise.resolve("a-instance"))
    return comp.find("Connect(Pagination)").props().onSelectRef("a-ref").then(() => {
      comp.update()
      expect(onSelectRef).toHaveBeenCalledWith("a-ref")
      expect(shallowToJson(comp)).toMatchSnapshot()
    })
  })

  it("does NOT display selected ref if no promise is returned", () => {
    onSelectRef.mockReturnValue(null)

    comp.find("Connect(Pagination)").simulate("selectRef", "a-ref")
    expect(onSelectRef).toHaveBeenCalledWith("a-ref")
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("can go back to instances list from instance details view", () => {
    comp.setState({ selectedInstance: "fake-instance" })
    expect(shallowToJson(comp)).toMatchSnapshot()

    comp.find("Button").simulate("click")
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("does not set onSelectRef for Pagination Component if no handler", () => {
    comp.setProps({ onSelectRef: null })
    expect(shallowToJson(comp)).toMatchSnapshot()
  })

  it("resets inner state when query function changes", () => {
    comp.setState({ selectedInstance: "a-instance" })
    comp.setProps({ query: () => null })
    expect(shallowToJson(comp)).toMatchSnapshot()
  })
})

