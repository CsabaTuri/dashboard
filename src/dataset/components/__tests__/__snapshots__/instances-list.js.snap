// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstancesList Component can go back to instances list from instance details view 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance="fake-instance"
    />
  </div>
</div>
`;

exports[`InstancesList Component can go back to instances list from instance details view 2`] = `
<div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance={null}
    />
  </div>
</div>
`;

exports[`InstancesList Component displays instances list 1`] = `
<div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance={null}
    />
  </div>
</div>
`;

exports[`InstancesList Component displays the selected instance 1`] = `
<div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance="a-instance"
    />
  </div>
</div>
`;

exports[`InstancesList Component does NOT display selected ref if no promise is returned 1`] = `
<div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance={null}
    />
  </div>
</div>
`;

exports[`InstancesList Component does not set onSelectRef for Pagination Component if no handler 1`] = `
<div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={null}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance={null}
    />
  </div>
</div>
`;

exports[`InstancesList Component resets inner state when query function changes 1`] = `
<div>
  <div
    style={
      Object {
        "display": "block",
      }
    }
  >
    <Connect(Pagination)
      onSelectRef={[Function]}
      query={[Function]}
    />
  </div>
  <div
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Button
      buttonType={4}
      icon="ChromeBack"
      onClick={[Function]}
    >
      Go back to the instances list
    </Button>
    <Connect(InstanceInfo)
      instance={null}
    />
  </div>
</div>
`;
